### PR TITLE
Feat: Ensure all features return built-in types

### LINKED ISSUE #1062: Features should return consistent types
The `NeuroM.features.*` functions return inconsistent types. For example, `segment_taper_rates` returns a list of `floats` while `section_taper_rates` returns a list of `numpy.float32`. This can be annoying to process results (e.g. it's harder to dump the results to JSON strings or stuff like that).
I suggest we make all the types consistent, like casting all the results to built-in Python types.
WDYT @eleftherioszisis @mgeplf ? If you agree I can work on this.

### PR DESCRIPTION
**Pull request by adrien-berchet**
Fixes #1062

### CODE REVIEW

File: tests/features/test_features.py [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+    The data should either be an int or float, or a list or tuple of ints or floats.
+    """
+    if isinstance(data, (list, tuple)):
+        for i in data:
+            _check_nested_type(i)
+    else:
+        assert isinstance(data, (int, float))
+
+
+class TestFeatureTypes:
```
Comment (eleftherioszisis) [inline tests/features/test_features.py:62]: Is the class necessary here? Could you list the tests without the class definition?

Comment (adrien-berchet) [reply tests/features/test_features.py:62]: It's not necessary, it's just to organize the tests. I can remove it if you prefer.

Comment (eleftherioszisis) [reply tests/features/test_features.py:62]: Yeah, I think that the tests are quite descriptive by themselves.

Comment (adrien-berchet) [reply tests/features/test_features.py]: Done

**Review by eleftherioszisis — APPROVED** [review]

**Pull request merged**